hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/source/StreamReadMonitoringFunction.java

```diff
@@ -177,6 +194,20 @@ public void initializeState(FunctionInitializationContext context) throws Except
           LOG.debug("{} retrieved an issued instant of time [{}, {}] for table {} with path {}.",
               getClass().getSimpleName(), issuedInstant, issuedOffset, conf.get(FlinkOptions.TABLE_NAME), path);
         }
+      } else if (retrievedStates.size() == 3) {
+        this.issuedInstant = retrievedStates.get(0);
+        this.issuedOffset = retrievedStates.get(1);
+        this.totalSplits = Integer.parseInt(retrievedStates.get(2));
```

Comment:
Is it necessary to make snapshot for `totalSplits`? it's kind of hacky to put it in `instantState`, which is used to store instant time.